### PR TITLE
Canonicalise unicode_language_id

### LINKED ISSUE #70: Canonicalise language, script, and region tags
Language, script, and region tags should be canonicalised, because
1. This matches how ECMA-402 works for other API.
1. ICU does this implicitly for some APIs resp. other ICU APIs require a canonicalised input to produce any result.

---

For example when `new Intl.DisplayNames("en", {type: "language"}).of("de-DD")` returns `"German (Germany)"`, for consistency we should then ensure that `new Intl.DisplayNames("en",{type: "region"}).of("DD")` returns `"Germany"`.

### PR DESCRIPTION
**Pull request by FrankYFTang**
Close #70

@anba @zbraniecki @sffc @leobalter @gibson042

### CODE REVIEW

File: spec.emu [hunk 1]
```diff
@@ -71,7 +71,7 @@ contributors: Google, Ecma International
       <emu-alg>
         1. If _type_ is `"language"`, then
           1. If _code_ does not matches the `unicode_language_id` production, throw a *RangeError* exceptipon.
-          1. Let _code_ be the result of mapping _code_ to lower case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
+          1. Set _code_ to CanonicalizeUnicodeLocaleId(_code_).
```
Comment (anba) [inline spec.emu:74]: `CanonicalizeUnicodeLocaleId` expects that its input is a valid string according to `IsStructurallyValidLanguageTag`, so the previous step should be replaced with `IsStructurallyValidLanguageTag`.

File: spec.emu [hunk 2]
```diff
@@ -71,7 +71,7 @@ contributors: Google, Ecma International
...
           1. Return _code_.
         1. If _type_ is `"region"`, then
           1. If _code_ does not matches the `unicode_region_subtag` production, throw a *RangeError* exceptipon.
```
Comment (anba) [inline spec.emu:77]: Region (and scripts) subtags should also get canonicalised to replaced outdated subtags with their preferred value.

**Review by anba — CHANGES_REQUESTED** [review]

File: spec.emu (hunk 1 above)
Comment (anba) [reply spec.emu:74]: Or rather we want to check if it matches `unicode_language_id` and `IsStructurallyValidLanguageTag` returns true, so locale ids with extension or private-use subtags aren't allowed.

File: spec.emu (hunk 2 above)
Comment (FrankYFTang) [reply spec.emu:77]: I have concern about this. (canonicalize the region code). There are no pre-defined process in UTS35 for this. The process for the region subtag within unicode_language_id stated in https://unicode-org.github.io/cldr/ldml/tr35.html#Canonical_Unicode_Locale_Identifiers depends on the language code (and script code if present) while there are multiple territories listed in the replacement attribute of territoryAlias.

File: spec.emu (hunk 1 above)
Comment (FrankYFTang) [reply spec.emu:74]: ok. Let me merge this PR AS IS first then we move on from there.

**Pull request merged**